src/main/java/org/folio/rest/impl/PrecedingSucceedingTitleAPI.java

```diff
@@ -68,6 +94,56 @@ public void putPrecedingSucceedingTitlesByPrecedingSucceedingTitleId(String prec
       PutPrecedingSucceedingTitlesByPrecedingSucceedingTitleIdResponse.class, asyncResultHandler);
   }
 
+  private boolean validatePrecedingSucceedingTitles(List<PrecedingSucceedingTitle> precedingSucceedingTitles,
+                                                    String instanceId,
+                                                    Handler<AsyncResult<Response>> asyncResultHandler) {
+    boolean areValidTitles = true;
+    for (PrecedingSucceedingTitle precedingSucceedingTitle : precedingSucceedingTitles) {
+      if (titleIsLinkedToInstanceId(precedingSucceedingTitle, instanceId)) {
+        var validationErrorMessage =
+          createValidationErrorMessage("precedingInstanceId or succeedingInstanceId", "",
+            String.format("The precedingInstanceId or succeedingInstanceId should contain instanceId [%s]", instanceId));
+        asyncResultHandler.handle(Future.succeededFuture(PutPrecedingSucceedingTitlesResponse
+          .respond422WithApplicationJson(validationErrorMessage)));
+        areValidTitles = false;
```

Comment:
If there are multiple invalid titles the asyncResultHandler is called multiple times resulting in an error because HTTP can return a single 422 message only.
Consider replacing `areValidTitles = false;` by `return false;` and `return areValidTitles;` by `return true;`

Reply:
Great catch. Thanks